### PR TITLE
fix(api,js): inbox api send workflow identifier

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
We've been missing sending the workflow identifier with the preferences response, this field is useful for the localization of the preferences.

### Screenshots

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Special notes for your reviewer

</details>

### CODE REVIEW

File: apps/api/src/app/inbox/inbox.controller.ts [hunk 1]
```diff
@@ -225,7 +225,7 @@ export class InboxController {
   async updateGlobalPreference(
     @SubscriberSession() subscriberSession: SubscriberEntity,
     @Body() body: UpdatePreferencesRequestDto
-  ): Promise<ISubscriberPreferences> {
+  ): Promise<InboxPreference> {
```
Comment (LetItRock) [inline apps/api/src/app/inbox/inbox.controller.ts:228]: created a new type for the preferences as the old one doesn't satisfy our needs and is used in other old NC use-cases

File: packages/js/src/ui/components/elements/Preferences/Preferences.tsx [hunk 2]
```diff
@@ -79,7 +76,7 @@ export const Preferences = () => {
             <PreferencesRow
               localizationKey={preference.workflow!.identifier as StringLocalizationKey}
               channels={preference.channels}
-              workflowId={preference.workflow?.identifier}
+              workflowId={preference.workflow?.id}
```
Comment (LetItRock) [inline packages/js/src/ui/components/elements/Preferences/Preferences.tsx:79]: fix for the preference update

File: packages/js/src/ui/components/elements/Preferences/Preferences.tsx [hunk 3]
```diff
@@ -45,10 +45,7 @@ export const Preferences = () => {
   }) => {
     mutate((prev) =>
       prev?.map((preference) => {
-        if (
-          preference.workflow?.identifier === workflowId ||
-          (!workflowId && preference.level === PreferenceLevel.GLOBAL)
-        ) {
+        if (preference.workflow?.id === workflowId || (!workflowId && preference.level === PreferenceLevel.GLOBAL)) {
```
Comment (LetItRock) [inline packages/js/src/ui/components/elements/Preferences/Preferences.tsx:48]: fix update by workflow id, not identifier

File: apps/api/src/app/inbox/utils/types.ts [hunk 4]
```diff
@@ -38,3 +38,18 @@ export type NotificationFilter = {
   read?: boolean;
   archived?: boolean;
 };
+
+export type Workflow = {
+  id: string;
+  identifier: string;
+  name: string;
+  critical: boolean;
+  tags?: string[];
```
Comment (BiswaViraj) [inline apps/api/src/app/inbox/utils/types.ts:47]: I think we should update the Workflow type in `js/src/types.ts` file too to add the tags

Comment (LetItRock) [reply apps/api/src/app/inbox/utils/types.ts:47]: thanks, will do! 👍

**Review by desiprisg — APPROVED** [review]

**Review by BiswaViraj — APPROVED** [review]

**Pull request merged**